tests/pipeline/test_factor.py

```diff
@@ -585,6 +586,40 @@ def test_returns(self, seed_value, window_length):
 
         check_allclose(expected, out)
 
+    @parameterized.expand([
+        (100, 15),
+        (101, 4),
+        (102, 100),
+        ])
+    def test_percentchange(self, seed_value, window_length):
+
+        pct_change = PercentChange(
+            inputs=[EquityPricing.close, ],
+            window_length=window_length,
+        )
+
+        today = datetime64(1, 'ns')
+        assets = arange(8)
+
+        seed(seed_value)  # Seed so we get deterministic results.
+        test_data = randn(window_length, 8)
```

Comment:
Another, more "functional" way to write this would be to use [np.concatenate](https://docs.scipy.org/doc/numpy/reference/generated/numpy.concatenate.html) or [np.vstack](https://docs.scipy.org/doc/numpy/reference/generated/numpy.vstack.html#numpy.vstack) to stack the first row, the middle row, and the end row together. (This is fine as is, but in other contexts concat might be clearer than updating the array in place.)